### ISSUE TITLE
vmi_init broken for Windows guests on KVM with Rekall profile

### ISSUE THREAD

**Issue by namidairo** [libvmi/libvmi#344]
vmi_init fails if you request VMI_INIT_COMPLETE when using a Rekall profile config on KVM, due to the KVM driver not handling it's request to retrieve the FS_BASE and/or GS_BASE. VMI_INIT_PARTIAL will still work if all you want is raw access.

The qemu monitor 'info registers' command, from which we are getting the rest of the registers looks like it should easily be extended to print out GS_BASE and FS_BASE too.

I think adding something along these lines in [target-1386/helper.c's x86_cpu_dump_state](https://github.com/qemu/qemu/blob/master/target-i386/helper.c#L422) should work, but it does mean extending the patches to Qemu for KVM users.

```
cpu_fprintf(f, 
"FS_BASE=%016" PRIx64 " " GS_BASE=%016 " PRIx64 "\n",
env->segs[R_FS].base,
env->segs[R_GS].base);
```

**Comment by tklengyel** [comment]
Good catch! Feel free to do a PR for updating the patch.

**Comment by namidairo** [comment]
Oh silly me, it does dump the segment registers with bases it seems.

`GS =002b fffff8032bf67000 ffffffff 00c0f300`
`FS =0053 000000007fea9000 00003c00 0040f300`

Which are the contents of this struct

```
typedef struct SegmentCache {
    uint32_t selector;
    target_ulong base;
    uint32_t limit;
    uint32_t flags;
} SegmentCache;
```

I'll still need to update the KVM driver to parse that though.

**Comment by namidairo** [comment]
Solved by #351
